src/closure_decorator.rs

```diff
@@ -294,47 +244,79 @@ impl<'a> VisitMut for ClosureDecorator<'a> {
         // )
         *expr = *self.register_mut_ast(&mut Expr::Fn(func.take()), ast);
       }
-      Expr::New(new) if new.args.is_some() => {
+      Expr::New(NewExpr { args, callee, .. }) if args.is_some() => {
         // detect if this looks like a new Proxy
         // e.g. `new Proxy({}, {})`
-        let maybe_proxy_class = match new.callee.as_mut() {
-          Expr::Ident(ident) if &ident.sym == "Proxy" => Some(ident),
-          _ => None,
-        };
+        let maybe_proxy_class = callee.as_mut_ident().filter(|ident| &ident.sym == "Proxy");
 
         if maybe_proxy_class.is_some() {
           let proxy_class = maybe_proxy_class.unwrap();
 
-          let args: Vec<ExprOrSpread> = iter::once(ExprOrSpread {
-            expr: Box::new(Expr::Ident(proxy_class.take())),
-            spread: None,
-          })
-          .chain(
-            new
-              .args
-              .as_mut()
-              .unwrap()
-              .iter_mut()
-              .map(|arg| ExprOrSpread {
-                spread: None,
-                expr: arg.expr.take(),
-              }),
-          )
-          .collect();
+          args
+            .iter_mut()
+            .for_each(|arg| arg.visit_mut_children_with(self));
+
+          let args = args
+            .as_mut()
+            .unwrap()
+            .iter_mut()
+            .map(|arg| ExprOrSpread {
+              spread: None,
+              expr: arg.expr.take(),
+            })
+            .collect();
```

Comment:
Combine?

Reply:
Will that work? I didn't think it would but will try.

Reply:
Got rid of the map completely